pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -37,18 +39,15 @@
     APPEND, INCREMENT, OVERWRITE, UNION
   }
 
+  @JsonPropertyDescription("Upsert mode.")
   private final Mode _mode;
-  private final Map<String, Strategy> _partialUpsertStrategies;
 
-  public UpsertConfig(@JsonProperty(value = "mode", required = true) Mode mode) {
```

Comment:
Can we just keep this constructor and make it call `this(mode, null)`? By doing so, you don't have to change all the places that invoke this method.

Reply:
I don't think changing the body of the constructor will do anything. The error was more about not being able to find a constructor with sufficient properties even though one was available.To avoid confusion, it's probably best to have only constructor with all properties. This class was the exception as other `BaseJsonConfig` derived classes are declaring only one constructor.